src/framework/lib/genesisframework/task.rb

```diff
@@ -112,32 +112,51 @@ def install provider, *what
               raise 'yum install exited with status: ' + $?.exitstatus.to_s
             end
           elsif provider == :gem
+            # convert arguments in "what" to a gem_name => [requires_list] structure
+            gems = {}
+            what.each { |item|
+              if item.is_a?(Hash)
+                gems.merge! item
+              else
+                gems[item] = [item]
+              end
+            }
+
             # We give a decent try at detecting if the gem is
             # installed before trying to reinstall again.
             # If it contains a - (aka you are specifying a specific version
             # or a / (aka you are specifying a path to find it) then
             # we punt on trying to determine if the gem is already
             # installed and just pass it to install anyway.
-            gems = what.select do |item|
-              item.include?("-") \
-                || item.include?("/") \
-                || Gem::Dependency.new(item).matching_specs.count == 0
+            install_gems = gems.select do |gem, requires|
+              gem.include?("-") \
+                || gem.include?("/") \
+                || Gem::Dependency.new(gem).matching_specs.count == 0
             end
-            if gems.size > 0    # make sure we still have something to do
-              Kernel.system('gem', 'install', '--no-ri', '--no-rdoc', *gems)
+
+            if install_gems.size > 0    # make sure we still have something to do
+              options = config.fetch(:gem_args, '').split
+              args = (options << install_gems.keys).flatten
+              Kernel.system('gem', 'install', *args)
               if $?.exitstatus != 0
-                raise "gem install #{gems.join(' ')} exited with status: " \
+                raise "gem install #{args.join(' ')} exited with status: " \
                   + $?.exitstatus.to_s
               end
             else                # be noisy that we aren't doing anything
-              puts "already installed gems: #{what.join(' ')}"
+              puts "already installed gems: #{gems.keys.join(' ')}"
             end
 
             # now need to clear out the Gem cache so we can load it
             Gem.clear_paths
 
-            # Now we require all the gems you asked to be installed
-            what.all? { |gem| require gem }
+            # Attempt to require loaded gems, print a message if we can't.
+            gems.each { |gem, requires|
+              begin
+                requires.each {|r| require r }
+              rescue LoadError
+                puts "Could not load gem #{gem} automatically. Maybe the gem name differs from its load path? Please specify the name to require, or load it explicitly in the task."
```

Comment:
Why is this a #puts instead of #raise ?  You don't handle/fix the error in any way so another error will just happen later and this message will probably be lost in the confusion.  You may want to #log this message too so it is visible in the asset's collins log.